### PR TITLE
Add Github Actions CI

### PR DESCRIPTION
**Pull request by palfrey**
Other fixes needed along the way
- Replaced pipenv with pip-tools generated requirements files.
- Corrected the error in `widgets.pyi`. This was mostly done because I couldn't get the regex support in pytest-mypy-plugins to work, which meant the test failed locally due to different source repo paths.
- Some flake8/black fixes in setup.py

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  publish:
```
Comment (DavisRayM) [inline .github/workflows/ci.yml]:
```suggestion
  ci:
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    branches:
+      - master
+
+jobs:
+  ci:
+    runs-on: ubuntu-20.04 # 22.04 doesn't have Python 3.6
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10", "3.11"]
```
Comment (DavisRayM) [inline .github/workflows/ci.yml]: We should probably drop support for 3.6; I don't see a need to support it for future versions.

**Review by DavisRayM — DISMISSED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (palfrey) [reply .github/workflows/ci.yml]: Sure, I can rip that out :)

Comment (palfrey) [reply .github/workflows/ci.yml]: Done

**Review by DavisRayM — APPROVED** [review]

**Pull request merged**